code/game/machinery/kitchen/smartfridge.dm

```diff
@@ -149,26 +165,66 @@
 		icon_state = icon_on
 	if(contents.len)
 		overlays += "drying_rack_filled"
-		if(!inoperable())
+		if(!inoperable() && currently_drying)
 			overlays += "drying_rack_drying"
 
 /obj/machinery/smartfridge/drying_rack/proc/dry()
+	var/drying_something = FALSE //While we're here, check if anything is undried and still processing
 	for(var/obj/item/weapon/reagent_containers/food/snacks/S in contents)
-		if(S.dry) continue
-		if(S.dried_type == S.type)
-			S.dry = 1
-			item_quants[S.name]--
-			S.name = "dried [S.name]"
-			S.color = "#AAAAAA"
-			S.loc = loc
+		if(S.dry)
+			continue
+		S.dryness += drying_power * (rand(0.85, 1.15))
+		if (S.dryness >= 1)
+			if(S.dried_type == S.type || !S.dried_type)
+				S.dry = 1
+				S.name = "dried [S.name]"
+				S.color = "#AAAAAA"
+			else
+				var/D = S.dried_type
+				D = new D(src)
+				if (istype(D, /obj/item/weapon/reagent_containers/food/snacks))
+					var/obj/item/weapon/reagent_containers/food/snacks/SD = D
+					SD.dry = TRUE //So we dont get stuck in an endless loop of drying, transforming and drying again
+				qdel(S)
+			update_contents()
 		else
-			var/D = S.dried_type
-			new D(loc)
-			item_quants[S.name]--
-			qdel(S)
-		return
+			drying_something = TRUE
+
+	if (drying_something != currently_drying)
+		currently_drying = drying_something
+		update_icon() //Only update the icon if we have to
+	currently_drying = drying_something
 	return
 
+
+
+
+
+
+
+
+
+
+
+
+/obj/machinery/smartfridge/New()
```

Comment:
Isn't it better to keep basic `/obj/machinery/smartfridge` together at file begining and subtypes of `/obj/machinery/smartfridge` and overrided procs at the end?

Reply:
But if you wanted to see how parent work you should be run up and down through the file to find different parts of code related to same type.
I think that way is cleaner:
```
- parent_type_defination
- parent_type_procs

- kind of space/separtator
- child_type_defintion
- child_type_procs
- repid last 3 steps if reqed
```

Reply:
I definitely agree that parent definition should be uptop, and all its vars, i just think child types should be below that, and all the main code and procs at the bottom.

For an example of why, I invite you to look at guns: https://i.imgur.com/uNBXf0z.png

All the different individual gun models either have their own file each, or are put into files with a couple other modules. These typically contain only variable definitions and occasionally a few special behaviours.
The core code that makes guns work is seperated too. People - especially new coders, often don't want to see that. They just come to these files to change the icon or description on their favourite gun.

Now we can't really do that for smartfridges, they're too niche to split over multiple files, it'd be a bit silly. But having the simple parts at the top is a good way to compromise. The complicated stuff is there if you want it. But you're less likely to want to look at that, than to look at name, icon, balancing values, etc